src/coreclr/jit/llvm.cpp

```diff
@@ -853,13 +883,34 @@ void buildCast(llvm::IRBuilder<>& builder, GenTreeCast* cast)
         Value* intValue = builder.CreateZExt(getGenTreeValue(cast->CastOp()).getValue(builder), getLLVMTypeForVarType(TYP_INT));
         mapGenTreeToValue(cast, intValue); // nothing to do except map the source value to the destination GenTree
     }
+    else if (cast->CastToType() == TYP_DOUBLE && cast->CastOp()->TypeIs(TYP_FLOAT))
+    {
+        mapGenTreeToValue(cast, builder.CreateFPCast(getGenTreeValue(cast->CastOp()).getValue(builder), getLLVMTypeForVarType(TYP_DOUBLE)));
+    }
+    else if (cast->CastToType() == TYP_INT && cast->CastOp()->TypeIs(TYP_FLOAT, TYP_DOUBLE))
+    {
+        mapGenTreeToValue(cast,
+            cast->IsUnsigned()
+                ? builder.CreateFPToUI(getGenTreeValue(cast->CastOp()).getValue(builder), getLLVMTypeForVarType(TYP_INT))
+                : builder.CreateFPToSI(getGenTreeValue(cast->CastOp()).getValue(builder), getLLVMTypeForVarType(TYP_INT)));
+    }
     else
     {
         // TODO: other casts
         failFunctionCompilation();
     }
 }
 
+void buildCnsDouble(llvm::IRBuilder<>& builder, GenTree* node)
```

Comment:
Use the strongly-typed `GenTreeCnsDbl*`?

Reply:
Thanks